bigquery_etl/cli/query.py

```diff
@@ -1262,6 +1265,151 @@ def render(name, sql_dir, output_dir):
             click.echo(rendered_sql)
 
 
+def _parse_partition_date(partition_date):
+    params = partition_date.split(":")
+    if len(params) != 3:
+        return None
+
+    # Check date format
+    try:
+        datetime.datetime.strptime(params[2], "%Y-%m-%d").date()
+    except ValueError:
+        return None
+
+    # Check column name
+    if re.match(r"^\w+$", params[0]):
+        return {params[0]: params[2]}
+
+
+def _validate_partition_date(ctx, param, partition_date):
+    """Process the CLI parameter check_date and set the parameter for BigQuery."""
+    # Will be None if launched from Airflow.  Also ctx.args is not populated at this stage.
+    if partition_date is None:
+        return None
+    else:
+        parsed = _parse_partition_date(partition_date)
+        if parsed is None:
```

Comment:
Should we consider raising this exception inside `_parse_partition_date` instead?

Reply:
No because the function is also called when parsing the ctx.args from Airflow and raising a click exception in that processing path is not appropriate.